libvast/src/type.cpp

```diff
@@ -622,6 +623,18 @@ bool congruent(const data& x, const type& y) {
   return visit(data_congruence_checker{}, y, x);
 }
 
+expected<void> replace_if_congruent(std::initializer_list<type*> xs,
+                                    const schema& with) {
+  for (auto x : xs)
+    if (auto w = with.find(x->name())) {
```

Comment:
While it's a bit more verbose, I would still prefer making use of the new C++17 if-initializer for readability. Also `t` (for *type*) instead of `w`:

```cpp
if (auto t = with.find(x->name(); t != nullptr) {
  // ...
}
```